src/cmake/thirdparty/SetupGeosxThirdParty.cmake

```diff
@@ -216,27 +216,17 @@ endif()
 if(DEFINED HDF5_DIR)
     message(STATUS "HDF5_DIR = ${HDF5_DIR}")
 
-    set(HDF5_ROOT ${HDF5_DIR})
-    set(HDF5_USE_STATIC_LIBRARIES FALSE)
-    set(HDF5_NO_FIND_PACKAGE_CONFIG_FILE ON)
-    include(FindHDF5)
-
-    # On some platforms (Summit) HDF5 lists /usr/include in it's list of include directories.
-    # When this happens you can get really opaque include errors.
-    list(REMOVE_ITEM HDF5_INCLUDE_DIRS /usr/include)
-
-    blt_import_library(NAME hdf5
-                       INCLUDES ${HDF5_INCLUDE_DIRS}
-                       LIBRARIES ${HDF5_LIBRARIES}
-                       TREAT_INCLUDES_AS_SYSTEM ON)
+    find_package(HDF5 REQUIRED
+                 PATHS ${HDF5_DIR}
+                 NO_DEFAULT_PATH)
 
-    file(READ "${HDF5_DIR}/include/H5public.h" header_file )
-    string(REGEX MATCH "version: *([0-9]+.[0-9]+.[0-9]+)" _ ${header_file})
-    set( HDF5_VERSION "${CMAKE_MATCH_1}" CACHE STRING "" FORCE )
     message( " ----> HDF5 version ${HDF5_VERSION}")
 
+    get_target_property(HDF5_INCLUDE_DIRS HDF5::HDF5 INTERFACE_INCLUDE_DIRECTORIES)
+    set_target_properties(HDF5::HDF5 PROPERTIES INTERFACE_SYSTEM_INCLUDE_DIRECTORIES "${HDF5_INCLUDE_DIRS}")
```

Comment:
Is it necessary to put the includes into system interface ?

Reply:
We do this so that they are included with `-isystem` instead of `-I` to suppress any compiler warnings.